src/compiler/transformers/utilities.ts

```diff
@@ -679,9 +679,9 @@ function getDecoratorsOfParameters(node: FunctionLikeDeclaration | undefined) {
  *
  * @internal
  */
-export function getAllDecoratorsOfClass(node: ClassLikeDeclaration): AllDecorators | undefined {
+export function getAllDecoratorsOfClass(node: ClassLikeDeclaration, useLegacyDecorators: boolean): AllDecorators | undefined {
     const decorators = getDecorators(node);
-    const parameters = getDecoratorsOfParameters(getFirstConstructorWithBody(node));
+    const parameters = useLegacyDecorators ? getDecoratorsOfParameters(getFirstConstructorWithBody(node)) : undefined;

```

Comment:
Should `getDecoratorsOfParameters` take the parameter and return undefined instead of copying the ternary to all call sites?

Reply:
Branching is cheaper at runtime than invocation, so probably not.

Reply:
In this case specifically, just passing the argument to `getDecoratorsOfParameters` requires iterating over all class elements to find a constructor, which is unnecessary work.